client/lib/url-search/index.js

```diff
@@ -2,37 +2,94 @@
  * External dependencies
  */
 import React from 'react';
+import debugModule from 'debug';
+import page from 'page';
+import url from 'url';
+import { pick } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { doSearch, getSearchOpen } from 'lib/mixins/url-search';
+const debug = debugModule( 'calypso:url-search' );
 
-export default function( Component ) {
-	const componentName = Component.displayName || Component.name || '';
+/**
+ * Function for constructing the url to page to. e.g.
+ * { uri: 'google.com', search:'hello' } --> 'google.com?s=hello'
+ *
+ * @param {Object} options the options object
+ * @param {string} options.uri the base uri to modify and add a query to
+ * @param {string} options.search the search term
+ * @param {string} [options.queryKey = s] the key to place in the url.  defaults to s
+ *
+ * @returns {string} The built search url
+ */
+const buildSearchUrl = ( { uri, search, queryKey = 's' } ) => {
+	const parsedUrl = pick(
+		url.parse( uri, true ),
+		'pathname',
+		'hash',
+		'query',
+	);
+
+	if ( search ) {
+		parsedUrl.query[ queryKey ] = search;
+	} else {
+		delete parsedUrl.query[ queryKey ];
```

Comment:
1. that is what current functionality is, and I want to not break the thing currently depending on it.
2. this component _is_ the thing setting the url param.  its supposed to be hooked up to a text component's change handler.  Without this line, the url param would never get unset even if you cleared your query